src/main/java/com/cloudbees/plugins/credentials/api/resource/APIResource.java

```diff
@@ -0,0 +1,47 @@
+package com.cloudbees.plugins.credentials.api.resource;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver;
+
+import java.util.Collections;
+import java.util.List;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.CUSTOM, property = "type", include = JsonTypeInfo.As.PROPERTY)
+@JsonTypeIdResolver(SymbolResolver.class)
+public class APIResource {
```

Comment:
I suggest you use a type parameter here so APIResource methods consume and produce homogeneous types from a programmatic point of view

Reply:
yeah, did not want to mess around with generics just for the PoC